db/version_set.cc

```diff
@@ -4321,9 +4369,7 @@ Status VersionSet::LogAndApplyHelper(ColumnFamilyData* cfd,
   edit->SetLastSequence(db_options_->two_write_queues ? last_allocated_sequence_
                                                       : last_sequence_);
 
-  Status s = builder->Apply(edit);
-
-  return s;
+  return builder ? builder->Apply(edit) : Status::OK();
```

Comment:
Maybe add an assertion or comment to explain why `Status::OK()` is returned when builder is nullptr.

Reply:
Sorry for not being specific.
I think under current circumstances, the following will be clearer.
```
// The builder can be nullptr only if edit is wal operation,
// and we return Status::OK() in this case.
assert(builder || edit->IsWalOperation());
```